src/PagerView.tsx

```diff
@@ -1,155 +1,207 @@
-    if (Platform.OS === 'android') {
-      if (this.props.initialPage != null) {
-        this.setPageWithoutAnimation(this.props.initialPage);
-      }
-    }
-  }
+type RenderWindowData = {
+  buffer: number | undefined;
+  currentPage: number;
+  maxRenderWindow: number | undefined;
+  offset: number;
+  windowLength: number;
+};
 
-  public getInnerViewNode = (): ReactElement => {
-    return this.PagerView.current!.getInnerViewNode();
-  };
-
-  private _onPageScroll = (e: PagerViewOnPageScrollEvent) => {
-    if (this.props.onPageScroll) {
-      this.props.onPageScroll(e);
-    }
-    // Not implemented on iOS yet
-    if (Platform.OS === 'android') {
-      if (this.props.keyboardDismissMode === 'on-drag') {
-        Keyboard.dismiss();
-      }
-    }
-  };
+export class PagerView<ItemT> extends React.PureComponent<
+  PagerViewProps<ItemT>,
+  PagerViewState
+> {
+  private isScrolling = false;
 
-  private _onPageScrollStateChanged = (
-    e: PageScrollStateChangedNativeEvent
-  ) => {
-    if (this.props.onPageScrollStateChanged) {
-      this.props.onPageScrollStateChanged(e);
-    }
-    this.isScrolling = e.nativeEvent.pageScrollState === 'dragging';
-  };
+  constructor(props: PagerViewProps<ItemT>) {
+    super(props);
+    this.state = this.computeRenderWindow({
+      buffer: props.buffer,
+      currentPage: props.initialPage ?? 0,
+      maxRenderWindow: props.maxRenderWindow,
+      offset: 0,
+      windowLength: 0,
+    });
+  }
 
-  private _onPageSelected = (e: PagerViewOnPageSelectedEvent) => {
-    if (this.props.onPageSelected) {
-      this.props.onPageSelected(e);
+  componentDidMount() {
+    if (this.props.initialPage != null && this.props.initialPage > 0) {
+      // Send command directly; render window already contains destination.
+      UIManager.dispatchViewManagerCommand(findNodeHandle(this), 'setPage', [
+        this.props.initialPage,
+        false,
+      ]);
     }
-  };
+  }
 
   /**
    * A helper function to scroll to a specific page in the PagerView.
-   * The transition between pages will be animated.
+   * Default to animated transition between pages.
    */
-  public setPage = (selectedPage: number) => {
-    UIManager.dispatchViewManagerCommand(
-      ReactNative.findNodeHandle(this),
-      getViewManagerConfig().Commands.setPage,
-      [selectedPage]
+  setPage(page: number, animated = true) {
+    if (page < 0 || page >= this.props.data.length) {
+      return;
+    }
+
+    // Start rendering the destination.
+    this.setState((prevState) =>
+      this.computeRenderWindow({
+        buffer: this.props.buffer,
+        currentPage: page,
+        maxRenderWindow: this.props.maxRenderWindow,
+        offset: prevState.offset,
+        windowLength: prevState.windowLength,
+      })
     );
-  };
+    // Send paging command.
+    requestAnimationFrame(() => {
+      UIManager.dispatchViewManagerCommand(findNodeHandle(this), 'setPage', [
```

Comment:
why 'setPage' instead of `getViewManagerConfig().Commands.setPage,`

Reply:
Was trying to work around https://github.com/facebook/react-native/commit/6eec39313dc6a088bc870a47de3a74c69714d4c9 .  However, perhaps that is Facebook internal only.  I can revert.